libs/langchain/tests/unit_tests/schema/runnable/test_history.py

```diff
@@ -0,0 +1,219 @@
+import datetime
+from typing import Callable, Sequence
+
+from langchain.memory import ChatMessageHistory
+from langchain.pydantic_v1 import BaseModel
+from langchain.schema import AIMessage, BaseMessage, HumanMessage
+from langchain.schema.runnable import RunnableConfig, RunnableLambda
+
+
+def get_factory() -> Callable:
+    chat_history_store = {}
+
+    def history_factory(session_id: str) -> ChatMessageHistory:
+        if session_id not in chat_history_store:
+            chat_history_store[session_id] = ChatMessageHistory()
+        return chat_history_store[session_id]
+
+    return history_factory
+
+
+def test_input_messages() -> None:
+    runnable = RunnableLambda(
+        lambda messages: "you said: "
+        + "\n".join(str(m.content) for m in messages if isinstance(m, HumanMessage))
+    )
+    factory = get_factory()
+    with_history = runnable.with_message_history(factory)
+    config: RunnableConfig = {"configurable": {"thread_id": datetime.datetime.now()}}
+    output = with_history.invoke([HumanMessage(content="hello")], config)
+    assert output == "you said: hello"
+    output = with_history.invoke([HumanMessage(content="good bye")], config)
+    assert output == "you said: hello\ngood bye"
+
+
+def test_input_dict() -> None:
+    runnable = RunnableLambda(
+        lambda input: "you said: "
+        + "\n".join(
+            str(m.content) for m in input["messages"] if isinstance(m, HumanMessage)
+        )
+    )
+    factory = get_factory()
+    with_history = runnable.with_message_history(factory, input_messages_key="messages")
+    config: RunnableConfig = {"configurable": {"thread_id": datetime.datetime.now()}}
```

Comment:
could we avoid using datetime.datetime.now() for a thread_id? can we hard-code the thread id to reasonable ids? like 1, 2, 3, 4 etc?